ecl/hqlcpp/hqlcse.cpp

```diff
@@ -292,26 +296,40 @@ void CseSpotter::analyseExpr(IHqlExpression * expr)
         extra->canAlias = true;
 
     bool savedCanAlias = canAlias;
-    if (expr->isDataset() && (op != no_select) && (!spotCseInIfDatasetConditions || (op != no_if)))
+    bool processed = false;
+    if (canAlias)
```

Comment:
note: This is primarily restructuring the code.  The only actual difference is for no_createset - which prevents expressions from the underlying dataset from being aliased (which caused problems in combination of some other changes)